GoogleForCreators/web-stories-wp
# Media: upload pre-existing poster image if available

## Context



## Summary

We're currently a bit wasteful with the poster generation, as we don't upload the poster to WordPress that we have already generated during upload, but instead generate a new one after upload completes.

## Relevant Technical Choices

By passing the poster file / blob around, it means that directly after an video is uploaded, the blob can be uploaded. This means there is no delay in regeneration and upload process is much smoother. 

## To-do



## User-facing changes



## Testing Instructions



### QA






This PR can be tested by following these steps:

#### Test with video optimization
1. Login.
2. Open editor. 
3. Upload video, via drag and drop in canvas / media library, copy and paste and via media picker. 
4. Await video to transcode and upload. 
5. Ensure poster image is set and upload. 

#### Test without video optimization
1. Login.
2. Stories > settings.
3. Disable video optimization 
4. Open editor. 
5. Upload video, via drag and drop in canvas / media library, copy and paste and via media picker. 
6. Await video to transcode and upload. 
7. Ensure poster image is set and upload. 

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6904
